modules/ldapbackend/dnssec.cc

```diff
@@ -0,0 +1,1385 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#include "exceptions.hh"
+#include "ldapbackend.hh"
+#include <cstdlib>
+
+
+bool LdapBackend::doesDNSSEC()
+{
+  return m_dnssec;
+}
+
+
+std::string LdapBackend::getDomainMetadataDN( const DNSName &name )
+{
+  std::string dn;
+  std::string filter = strbind( ":domain:", m_pldap->escape( name.toStringRootDot() ), getArg( "metadata-searchfilter" ) );
+  const char* attributes[] = { "objectClass", NULL };
+  PowerLDAP::sentry_t result;
+
+  try {
+    PowerLDAP::SearchResult::Ptr search = m_pldap->search( m_metadata_searchdn, LDAP_SCOPE_SUBTREE, filter, attributes );
+
+    if ( search->getNext( result, true ) ) {
+      if ( result.count( "dn" ) && !result["dn"].empty() )
+        dn = result["dn"][0];
+    }
+
+    L<<Logger::Debug<< m_myname << " getDomainMetadataDN will return " << dn << endl;
+    return dn;
+  }
+  catch( LDAPTimeout &lt )
+  {
+    L << Logger::Warning << m_myname << " Unable to search LDAP directory: " << lt.what() << endl;
+    throw( DBException( "LDAP server timeout" ) );
+  }
+  catch( LDAPNoConnection &lnc )
+  {
+    L << Logger::Warning << m_myname << " Connection to LDAP lost, trying to reconnect" << endl;
+    if ( reconnect() )
+      return this->getDomainMetadataDN( name );
+    else
+      throw PDNSException( "Failed to reconnect to LDAP server" );
+  }
+  catch( LDAPException &le )
+  {
+    L << Logger::Error << m_myname << " Unable to search LDAP directory: " << le.what() << endl;
+    throw( PDNSException( "LDAP server unreachable" ) );   // try to reconnect to another server
+  }
+  catch( std::exception &e )
+  {
+    L << Logger::Error << m_myname << " Caught STL exception searching for domain " << name << " under the metadata DN: " << e.what() << endl;
+    throw( DBException( "STL exception" ) );
+  }
+}
+
+
+bool LdapBackend::getDomainMetadata( const DNSName& name, const std::string& kind, std::vector<std::string>& meta )
+{
+  L<<Logger::Debug<< m_myname << " Getting metadata " << kind << " for domain " << name << endl;
+
+  if ( !m_dnssec && isDnssecDomainMetadata( kind ) )
+    return false;
+
+  std::string basedn = this->getDomainMetadataDN( name );
+  if ( basedn.empty() )
+    return false;
+  std::string filter = "(&(objectClass=PdnsMetadata)(cn=" + m_pldap->escape( kind ) + "))";
+  const char* attributes[] = { "PdnsMetadataValue", NULL };
+  PowerLDAP::sentry_t result;
+
+  try {
+    PowerLDAP::SearchResult::Ptr search = m_pldap->search( basedn, LDAP_SCOPE_SUBTREE, filter, attributes );
+
+    // We're supposed to get just one entry
+    if ( search->getNext( result, m_getdn ) ) {
+      if ( result.count( "PdnsMetadataValue" ) && !result["PdnsMetadataValue"].empty() ) {
+        for ( auto value : result["PdnsMetadataValue"] )
```

Comment:
1) i imagine this could be const auto&
2) curly braces for the for body would be appreciated for readability (too many levels of indent...)